src/Uno.Toolkit.UI/Controls/ResponsiveView/ResponsiveView.cs

```diff
@@ -0,0 +1,145 @@
+	#endregion
+
+	#region Normal DP
+	public DataTemplate NormalContent
+	{
+		get { return (DataTemplate)GetValue(NormalContentProperty); }
+		set { SetValue(NormalContentProperty, value); }
+	}
+
+	public static readonly DependencyProperty NormalContentProperty =
+		DependencyProperty.Register("NormalContent", typeof(DataTemplate), typeof(ResponsiveView), new PropertyMetadata(null));
+	#endregion
+
+	#region Wide DP
+	public DataTemplate WideContent
+	{
+		get { return (DataTemplate)GetValue(WideContentProperty); }
+		set { SetValue(WideContentProperty, value); }
+	}
+
+	public static readonly DependencyProperty WideContentProperty =
+		DependencyProperty.Register("WideContent", typeof(DataTemplate), typeof(ResponsiveView), new PropertyMetadata(null));
+	#endregion
+
+	#region Widest DP
+	public DataTemplate WidestContent
+	{
+		get { return (DataTemplate)GetValue(WidestContentProperty); }
+		set { SetValue(WidestContentProperty, value); }
+	}
+
+	public static readonly DependencyProperty WidestContentProperty =
+		DependencyProperty.Register("WidestContent", typeof(DataTemplate), typeof(ResponsiveView), new PropertyMetadata(null));
+	#endregion
+
+	#region ResponsiveLayout DP
+	public static DependencyProperty ResponsiveLayoutProperty { get; } = DependencyProperty.Register(
+		nameof(ResponsiveLayout),
+		typeof(ResponsiveLayout),
+		typeof(ResponsiveView),
+		new PropertyMetadata(default));
+
+	public ResponsiveLayout ResponsiveLayout
+	{
+		get => (ResponsiveLayout)GetValue(ResponsiveLayoutProperty);
+		set => SetValue(ResponsiveLayoutProperty, value);
+	}
+	#endregion
+
+	#endregion
+
+	public ResponsiveView()
+	{
+		this.DefaultStyleKey = typeof(ResponsiveView);
+
+		ResponsiveHelper.GetForCurrentView().Register(this);
+
+		Loaded += ResponsiveView_Loaded;
+	}
+
+	private void ResponsiveView_Loaded(object sender, RoutedEventArgs e)
+	{
+		var contentToSet = GetInitialValue();
+
+		SetContent(contentToSet);
+	}
+
+	private DataTemplate? GetInitialValue()
+	{
+		var helper = ResponsiveHelper.GetForCurrentView();
+
+		return GetValueForSize(helper.WindowSize, ResponsiveLayout ?? helper.Layout);
+	}
+
+	private DataTemplate? GetValueForSize(Size size, ResponsiveLayout layout)
+	{
+		var defs = new (double MinWidth, DataTemplate? Value)?[]
+		{
+			(layout.Narrowest, NarrowestContent),
+			(layout.Narrow, NarrowContent),
+			(layout.Normal, NormalContent),
+			(layout.Wide, WideContent),
+			(layout.Widest, WidestContent),
+		}.Where(x => x?.Value != null).ToArray();
+
+		var match = defs.FirstOrDefault(y => y?.MinWidth >= size.Width) ?? defs.LastOrDefault();
+
+		return match?.Value;
+	}
+
+	public void OnSizeChanged(Size size, ResponsiveLayout layout)
+	{
+		var contentToSet = GetValueForSize(size, ResponsiveLayout ?? layout);
+
+		SetContent(contentToSet);
+	}
+
+	private void SetContent(DataTemplate? contentToSet)
+	{
+		if (contentToSet is not null)
+		{
+			Content = contentToSet.LoadContent() as UIElement;
+		}
+	}
```

Comment:
just set the content-template directly
and also, no need a null guard, we actually want this to clear if no match
```suggestion
	public void OnSizeChanged(Size size, ResponsiveLayout layout)
	{
		ContentTemplate = GetValueForSize(size, ResponsiveLayout ?? layout);
	}
```

Reply:
Done in my next commits